vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/SpaceManager.java

```diff
@@ -138,4 +142,24 @@ public ISpace getFullyLoadedSpace(String id) {
         spaceRepo.findAll().forEach(s -> spaces.add(s));
         return spaces;
     }
+    
+    /**
+     * Method to delete space based on id
+     * 
+     * @param id  if id is null or space with input id is not found return corresponding error message, else delete and return success message
+     * @return response message  
+     */
+    @Override
+    public CustomResponses deleteSpaceById(String id) {
+        if(id!=null) {
+            if(!spaceRepo.findById(id).isPresent()){
+                return CustomResponses.NOT_FOUND;
+            } else {
+                spaceRepo.deleteById(id);
+                return CustomResponses.SUCCESS;
+            }
+        } else {
+            return CustomResponses.NULL_INPUT;
+        }
```

Comment:
this method does not need to return anything. if someone tries to delete a space that's not there, that's fine. just don't do anything. if it was successfully deleted, great, don't return anything. if id is null, just throw an exception since it clearly is an invalid input.